[KHRGA-86] Updated EP h_item classes (#113)
KhronosGroup/SYCL_Reference: source/iface/h_item.rst

::

  template <int Dimensions>
  class h_item;

> ``h_item<int Dimensions>`` identifies an instance of a

Reviewer: ```suggestion
The ``h_item`` class identifies an instance of a
```

``group::parallel_for_work_item`` function object executing at each

> point in a local ``range<int Dimensions>`` passed to a

Reviewer: ```suggestion
point in a local ``range`` passed to a
```

Author: Fixed with commit https://github.com/KhronosGroup/SYCL_Reference/pull/113/commits/3a84403759688916ba97a08d272b7841cd398783

``parallel_for_work_item`` call or to the corresponding
``parallel_for_work_group`` call if no ``range`` is passed to the
``parallel_for_work_item`` call. It encapsulates enough information
to identify the work-item's local and global items according to the
information given to ``parallel_for_work_group`` (physical ids) as well
as the work-item's logical local items in the logical local range.

> All returned items objects are offset-less. Instances of the

Reviewer: ```suggestion
All returned item objects are offset-less. Instances of the
```

Author: Fixed with commit https://github.com/KhronosGroup/SYCL_Reference/pull/113/commits/3a84403759688916ba97a08d272b7841cd398783

> ``h_item<int Dimensions>`` class are not user-constructible and

Reviewer: ```suggestion
``h_item`` class are not user-constructible and
```

  Based on developer and implementation feedback, the hierarchical
  data parallel kernel feature is undergoing improvements to better
  align with the frameworks and patterns prevalent in modern programming.

  As this is a key part of the SYCL API and we expect to make changes to
  it, we temporarily recommend that new codes refrain from using this
  feature until the new API is finished in a near-future version of the
  SYCL specification, when full use of the updated feature will be
  recommended for use in new code.

  Existing codes using this feature will of course be supported by
  conformant implementations of this specification.

.. seealso:: |SYCL_SPEC_H_ITEM|

================
Member functions
================

``get_global``
==============

::

  item<Dimensions, false> get_global() const

Return the constituent global item representing the work-item’s position
in the global iteration space as provided upon kernel invocation.

``get_local``
=============

::

  item<Dimensions, false> get_local() const

Return the same value as ``get_logical_local()``.

``get_logical_local``
=====================

::

  item<Dimensions, false> get_logical_local() const

> Return the constituent element of the logical local item work-item’s

Reviewer: ```suggestion
Return the constituent element of the logical local work-item’s
```

Author: Fixed with commit https://github.com/KhronosGroup/SYCL_Reference/pull/113/commits/3a84403759688916ba97a08d272b7841cd398783

position in the local iteration space as provided upon the invocation
of the ``group::parallel_for_work_item``.

> If the ``group::parallel_for_work_item`` was called without any logical

Reviewer: ```suggestion
If ``group::parallel_for_work_item`` was called without any logical
```

Author: Fixed with commit https://github.com/KhronosGroup/SYCL_Reference/pull/113/commits/3a84403759688916ba97a08d272b7841cd398783

local range then the member function returns the physical local item.

A physical id can be computed from a logical id by getting the remainder
of the integer division of the logical id and the physical range:
``get_logical_local().get() % get_physical_local.get_range() ==
get_physical_local().get()``.

``get_physical_local``
======================

::

  item<Dimensions, false> get_physical_local() const

> Return the constituent element of the physical local item work-item’s

Reviewer: ```suggestion
Return the constituent element of the physical local work-item’s
```

Author: Fixed with commit https://github.com/KhronosGroup/SYCL_Reference/pull/113/commits/3a84403759688916ba97a08d272b7841cd398783
